From d536ce198d4db1975125b6ab3d32f1bd076e79b2 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Fri, 15 May 2020 15:15:05 +0800
Subject: [PATCH] Revert "drm/rockchip: Convert MCU cmd from rgb565/rgb666 to
 rgb888"

This reverts commit 509d945c0ef72974cc18f0466cb4e6577a3c6c0e.

Use 79e05b456627("drm/rockchip: vop: Set mcu mode before setting output
mode and dither") as a better workaround.

Change-Id: I0e7fb791d9e14fb19e8f44c7d52eaf11d5d4401c
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop.c | 24 ---------------------
 1 file changed, 24 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
index 18c819da64b2..17181544dcef 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
@@ -2657,36 +2657,12 @@ static void vop_crtc_close(struct drm_crtc *crtc)
 
 static void vop_crtc_send_mcu_cmd(struct drm_crtc *crtc,  u32 type, u32 value)
 {
-	struct rockchip_crtc_state *s;
 	struct vop *vop = NULL;
 
 	if (!crtc)
 		return;
 
 	vop = to_vop(crtc);
-	s = to_rockchip_crtc_state(crtc->state);
-
-	/*
-	 * VOP wrongly treated MCU cmd as normal rgb data and pass it
-	 * to dither module when output mode is rgb565/rgb666, then
-	 * the cmd output from vop io is changed.
-	 *
-	 * Here we convert the MCU cmd data(8bit) from rgb565/rgb666 to rgb88,
-	 * so that we can get the original cmd data after dither module.
-	 * RGB565->RGB888:
-	 * B = cmd & 0x1f; G = (cmd & 0xe0) >> 5; R = 0
-	 * B + (G << (8 + (8 - 6))) + R
-	 * RGB666->RGB888:
-	 * B = cmd & 0x3f; G = (cmd & 0xc0) >> 6; R = 0
-	 * B + (G << (8 + (8 - 6))) + R
-	 *
-	 * Note: cmd is 8 bit, so R is always zero;
-	 */
-	if (s->output_mode == ROCKCHIP_OUT_MODE_P565)
-		value = ((value & 0x1f) << 3) + ((value & 0xe0) << 5);
-	else if (s->output_mode == ROCKCHIP_OUT_MODE_P666)
-		value = ((value & 0x3f) << 2) + ((value & 0xc0) << 4);
-
 	mutex_lock(&vop->vop_lock);
 	if (vop && vop->is_enabled) {
 		switch (type) {
-- 
2.35.3

